catalog/spatial/registry/registry-publication-action-provider/src/main/java/org/codice/ddf/registry/publication/action/provider/RegistryPublicationActionProvider.java

```diff
@@ -0,0 +1,262 @@
+
+    private static final Logger LOGGER =
+            LoggerFactory.getLogger(RegistryPublicationActionProvider.class);
+
+    private static final String REGISTRY_PATH = "/registries";
+
+    private static final String PUBLICATION_PATH = "publication";
+
+    private static final String PUBLISH_DESCRIPTION = "Publishes this nodes information to ";
+
+    private static final String PUBLISH_OPERATION = "publish";
+
+    private static final String PUBLISH_TITLE = "Publish to ";
+
+    private static final String UNPUBLISH_DESCRIPTION = "Unpublishes this nodes information from ";
+
+    private static final String UNPUBLISH_OPERATION = "unpublish";
+
+    private static final String UNPUBLISH_TITLE = "Unpublish from ";
+
+    private static final String HTTP_POST = "HTTP_POST";
+
+    private static final String HTTP_DELETE = "HTTP_DELETE";
+
+    private static final String METACARD_PROPERTY = "ddf.catalog.event.metacard";
+
+    private static final String CREATED_TOPIC = "ddf/catalog/event/CREATED";
+
+    private static final String UPDATED_TOPIC = "ddf/catalog/event/UPDATED";
+
+    private static final String DELETED_TOPIC = "ddf/catalog/event/DELETED";
+
+    private String providerId;
+
+    private ConfigurationAdmin configAdmin;
+
+    private FederationAdminService federationAdminService;
+
+    private List<RegistryStore> registryStores;
+
+    private Map<String, List<String>> publications = new ConcurrentHashMap<>();
+
+    @Override
+    public <T> List<Action> getActions(T subject) {
+        String registryId = getRegistryId(subject);
+        if (registryId == null) {
+            return Collections.emptyList();
+        }
+        List<Action> actions = new ArrayList<>();
+        List<String> currentPublications = publications.get(registryId);
+
+        for (RegistryStore registry : registryStores) {
+            //can't publish to yourself
+            if (!registry.isPushAllowed() || StringUtils.isBlank(registry.getRegistryId())
+                    || registryId.equals(registry.getRegistryId())) {
+                continue;
+            }
+
+            CollectionUtils.addIgnoreNull(actions,
+                    getAction(registryId,
+                            registry.getId(),
+                            !currentPublications.contains(registry.getId())));
+        }
+
+        return actions;
+    }
+
+    @Override
+    public String getId() {
+        return providerId;
+    }
+
+    @Override
+    public <T> boolean canHandle(T subject) {
+        return getRegistryId(subject) != null;
+    }
+
+    @Override
+    public void handleEvent(Event event) {
+        Metacard mcard = (Metacard) event.getProperty(METACARD_PROPERTY);
+        if (mcard == null || !mcard.getTags()
+                .contains(RegistryConstants.REGISTRY_TAG)) {
+            return;
+        }
+
+        String registryId = mcard.getAttribute(RegistryObjectMetacardType.REGISTRY_ID)
+                .getValue()
```

Comment:
I was just talking about this with @clockard we basically have several areas getting the registry ID attribute from the metacard without checking it. We should see if we can make a common method to obtain the registryID and check for null since getAttribute can return null. That would be an improvement to the registry related code.